src/Chainweb/Miner/Coordinator.hs

```diff
@@ -256,18 +259,33 @@ mineCut logfun conf nid cdb g !c !adjustments = do
     blockDb :: ChainId -> Maybe BlockHeaderDb
     blockDb cid = wcdb ^? webBlockHeaderDb . ix cid
 
+    -- TODO What's with the discrepancy between `as` and `adjustments` here?
+    -- Which is the true container we wish to manipulate?
     getTarget
-        :: ChainId
+        :: ChainwebVersion
+        -> ChainId
         -> BlockHeader
         -> Adjustments
         -> IO (T2 HashTarget Adjustments)
-    getTarget cid bh as = case HM.lookup (_blockHash bh) as of
-        Just (T2 _ t) -> pure $! T2 t adjustments
-        Nothing -> case blockDb cid of
+    getTarget v cid bh as = case v of
+        Test{} -> testTarget
+        TimedConsensus{} -> testTarget
+        TimedCPM{} -> testTarget
+        PowConsensus{} -> prodTarget
+        Development -> prodTarget
+        Testnet00 -> prodTarget
+        Testnet01 -> prodTarget
+        Testnet02 -> prodTarget
```

Comment:
could we move this case distinction into into `Chainweb.Version`? E.g. using something like

```Haskell
data TargetType = ProcTarget | TestTarget

getTargetType :: ChainwebVersion -> TargetType
```